cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -275,67 +274,91 @@ angular.module(PKG.name + '.commons')
     }
 
     function initNodes() {
-      vm.instance.unmakeEverySource();
-      vm.instance.unmakeEveryTarget();
-
-      // This is to handle special case when we open a pipeline with only one node,
-      // thought not sure why jsPlumb behaves differently in this case. Exclude
-      // condition nodes since we add endpoints differently for those
-      if ($scope.nodes.length === 1 && $scope.nodes[0].type !== 'condition') {
-        vm.instance.deleteEveryEndpoint();
-      }
-
       angular.forEach($scope.nodes, function (node) {
         if (node.type === 'condition') {
           initConditionNode(node.name);
         } else if (node.type === 'splittertransform') {
           initSplitterNode(node);
         } else {
-          initNormalNode(node.name);
+          initNormalNode(node);
         }
 
-        vm.instance.makeTarget(node.name, {
-          isTarget: true,
-          dropOptions: { hoverClass: 'drag-hover' },
-          anchor: 'ContinuousLeft',
-          allowLoopback: false
-        });
+        if (!vm.instance.isTarget(node.name)) {
+          let targetOptions = {
```

Comment:
Should we get this from my-dag-factory? we have defined a bunch of jsplumb specific settings in the factory.